### PR TITLE
Add item selection and manipulation methods to ndarray

### PR DESCRIPTION
**Pull request by sproshev**
Please don't merge until I add tests :)

### CODE REVIEW

**Comment by sproshev** [comment]
I'll update this PR tomorrow because I found some mistakes while writing tests,

File: numpy-stubs/__init__.pyi [hunk 1]
```diff
@@ -387,6 +394,58 @@ class ndarray(_ArrayOrScalarCommon, Iterable, Sized, Container):
     def ravel(self, order: str=...) -> ndarray: ...
     def squeeze(self, axis: Union[int, Tuple[int, ...]]=...) -> ndarray: ...
 
+    # Item selection and manipulation
+    @overload
+    def take(self, indices: int, *, mode: str=...) -> Any: ...
+    @overload
+    def take(self, indices: int, axis: int, out: ndarray=..., mode: str=...) -> Any: ...
+    @overload
+    def take(self, indices: Sequence[int], axis: int=..., out: ndarray=..., mode: str=...) -> ndarray: ...
```
Comment (shoyer) [inline numpy-stubs/__init__.pyi]: `indices` can take array-like arguments, not just `Sequence`.

Does `np.ndarray` pass `Sequence` checks?

File: numpy-stubs/__init__.pyi [hunk 2]
```diff
@@ -29,11 +30,17 @@ if sys.version_info[0] < 3:
...
+# Used to type hint values that could be in `ndarray`
+_T = TypeVar("_T")
+
 _Shape = Tuple[int, ...]
 
 # Anything that can be coerced to a shape tuple
 _ShapeLike = Union[int, Sequence[int]]
 
+# Anything that mentioned as `array_like ` in  documentation
+_ArrayLike = Union[_T, Sequence[_T]]
```
Comment (shoyer) [inline numpy-stubs/__init__.pyi]: I support the idea of clearly document NumPy's interface, but in practice, how does this specific annotation differ from `Any`?

I get the idea of trying to constrain to only "array-like" inputs, but keep in mind that for NumPy that means practically anything -- we can coerce almost any type of object to ndarray with `np.array()`.

File: numpy-stubs/__init__.pyi (hunk 1 above)
Comment (sproshev) [reply numpy-stubs/__init__.pyi]: Oh! Great finding, `ndarray` has all `Sequence`'s methods so I'll specify `Sequence` as a super-class

File: numpy-stubs/__init__.pyi (hunk 2 above)
Comment (sproshev) [reply numpy-stubs/__init__.pyi]: I was trying to prepare `ndarray` to become generic and limit passed argument to have same `dtype` as `self` but after a bit more testing I agree with you. It's better to put `Any` here.

**Comment by person142** [comment]
Closing; but feel free to make a new PR against the NumPy main repo!

**Pull request closed without merging**